presto-native-tests/src/test/java/com.facebook.presto.nativetests/AbstractTestQueriesNative.java

```diff
@@ -0,0 +1,1208 @@
+    }
+
+    @Override
+    @Test
+    public void testApproxSetGroupByWithNulls()
+    {
+        MaterializedResult actual = computeActual("" +
+                "SELECT orderstatus, cardinality(approx_set(IF(custkey % 2 <> 0, custkey))) " +
+                "FROM orders " +
+                "GROUP BY orderstatus");
+
+        MaterializedResult expected = resultBuilder(getSession(), actual.getTypes())
+                .row("O", 499L)
+                .row("F", 496L)
+                .row("P", 153L)
+                .build();
+
+        assertEqualsIgnoreOrder(actual.getMaterializedRows(), expected.getMaterializedRows());
+    }
+
+    @Override
+    @Test
+    public void testApproxSetVarchar()
+    {
+        MaterializedResult actual = computeActual("SELECT cardinality(approx_set(CAST(custkey AS VARCHAR))) FROM orders");
+
+        MaterializedResult expected = resultBuilder(getSession(), BIGINT)
+                .row(1015L)
+                .build();
+
+        assertEquals(actual.getMaterializedRows(), expected.getMaterializedRows());
+    }
+
+    @Override
+    @Test
+    public void testApproxSetVarcharGroupBy()
+    {
+        MaterializedResult actual = computeActual("" +
+                "SELECT orderstatus, cardinality(approx_set(CAST(custkey AS VARCHAR))) " +
+                "FROM orders " +
+                "GROUP BY orderstatus");
+
+        MaterializedResult expected = resultBuilder(getSession(), actual.getTypes())
+                .row("O", 1012L)
+                .row("F", 1011L)
+                .row("P", 304L)
+                .build();
+
+        assertEqualsIgnoreOrder(actual.getMaterializedRows(), expected.getMaterializedRows());
+    }
+
+    @Override
+    @Test
+    public void testApproxSetWithNulls()
+    {
+        MaterializedResult actual = computeActual("SELECT cardinality(approx_set(IF(orderstatus = 'O', custkey))) FROM orders");
+
+        MaterializedResult expected = resultBuilder(getSession(), actual.getTypes())
+                .row(1003L)
+                .build();
+
+        assertEquals(actual.getMaterializedRows(), expected.getMaterializedRows());
+    }
+
+    @Override
+    @Test
+    public void testApproxSetGroupByWithOnlyNullsInOneGroup()
+    {
+        MaterializedResult actual = computeActual("" +
+                "SELECT orderstatus, cardinality(approx_set(IF(orderstatus != 'O', custkey))) " +
+                "FROM orders " +
+                "GROUP BY orderstatus");
+
+        MaterializedResult expected = resultBuilder(getSession(), actual.getTypes())
+                .row("O", null)
+                .row("F", 1001L)
+                .row("P", 304L)
+                .build();
+
+        assertEqualsIgnoreOrder(actual.getMaterializedRows(), expected.getMaterializedRows());
+    }
+
+    @Override
+    @Test
+    public void testArrayCumSumDecimals()
+    {
+        String functionNotRegisteredError = ".*Scalar function presto.default.array_cum_sum not registered with arguments.*";
```

Comment:
Do we have this error because we don't have this functionality ? I don't see a reason for not building it. Do we have an issue for this ? Please create one if its not there and add the link here.